Orange/data/table.py

```diff
@@ -1388,9 +1595,15 @@ def _filter_has_class(self, negate=False):
                 retain = (self._Y.indptr[1:] ==
                           self._Y.indptr[-1:] + self._Y.shape[1])
         else:
-            retain = bn.anynan(self._Y, axis=1)
+            if self._Y.ndim == 1:
+                retain = np.isnan(self._Y)
+            else:
+                retain = bn.anynan(self._Y, axis=1)
             if not negate:
                 retain = np.logical_not(retain)
+        # TODO: Decide whether to keep this or not;
+        if np.all(retain):
+            return self
```

Comment:
@markotoplak, a similar question like above. Add this shortcut or not? It does not save any significant time or space, it just avoids creating an (non-unlockable) view. As far as I recall, this change is not needed with the current version of code, but may be useful. (Though: if somebody calls a filter, (s)he cannot count on getting an unlockable table, becuase this will still depend upon whether the filter kept all data or not.)

Reply:
I like the shortcut.